cylc/uiserver/resolvers.py

```diff
@@ -379,10 +379,14 @@ async def cat_log(cls, id_: Tokens, log, info, file=None):
                     if buffer:
                         yield {'lines': list(buffer)}
                         buffer.clear()
-                    if proc.returncode not in [None, 0]:
-                        (_, stderr) = await proc.communicate()
-                        # pass any error onto ui
-                        yield {'error': stderr.decode()}
+                    if proc.returncode is not None:
+                        if proc.returncode:
+                            (_, stderr) = await proc.communicate()
+                            # pass any error onto ui
+                            yield {'error': stderr.decode()}
+                        else:
+                            # cat-log exited 0 meaning file was removed
```

Comment:
Btw something I found out with the functional tests for https://github.com/cylc/cylc-flow/pull/5559 is that on the Ubuntu (GH Actions) version of `tail`, it exits 1 when the file is deleted. Whereas on our version of `tail` it exits 0. Anyway, both cases are handled here.

Reply:
Does tail exiting non-zero cause cat-log to exit non-zero?

Reply:
Oh, just checked and yes it does. `cat-log` exits with whatever `tail` exited with

Reply:
Looks like the exit code is filesystem dependent for the file removed case, possibly due to the presence/absence of inotify.